src/api/checks.h

```diff
@@ -201,6 +201,103 @@ namespace api {
     }                                                                          \
   } while (0)
 
+/* -------------------------------------------------------------------------- */
+/* Term checks.                                                               */
+/* -------------------------------------------------------------------------- */
+
+/**
+ * Term check for member functions of classes other than class Solver.
+ * Check if given term is not null and associated with the solver object this
+ * object is associated with.
+ */
+#define CVC4_API_CHECK_TERM(term)            \
+  do                                         \
+  {                                          \
+    CVC4_API_ARG_CHECK_NOT_NULL(term);       \
+    CVC4_API_ARG_CHECK_SOLVER("term", term); \
+  } while (0)
+
+/**
+ * Term check for member functions of classes other than class Solver.
+ * Check if each term in the given container of terms is not null and
+ * associated with the solver object this object is associated with.
+ */
+#define CVC4_API_CHECK_TERMS(terms)                                            \
+  do                                                                           \
+  {                                                                            \
+    size_t i = 0;                                                              \
+    for (const auto& s : terms)                                                \
+    {                                                                          \
+      CVC4_API_ARG_AT_INDEX_CHECK_NOT_NULL("term", s, terms, i);               \
+      CVC4_API_ARG_AT_INDEX_CHECK_EXPECTED(                                    \
+          this->d_solver == s.d_solver, "term", terms, i)                      \
+          << "a term associated with the solver this term is associated with"; \
```

Comment:
```suggestion
          << "a term associated with the solver this object is associated with"; \
```
It would be great to pass and print the name of the class as well.

Reply:
I'd rather not pass the class name as an argument unless we really have to -- and there's no nice automatic way of doing this....

Reply:
I mean, I can introduce macros per class, that's one thing I wanted to avoid, but if we want to get the class name in there it actually might make sense.

Reply:
> I mean, I can introduce macros per class, that's one thing I wanted to avoid, but if we want to get the class name in there it actually might make sense.

I would prefer not to print the class name if it comes to that 😐